src/agent/solana/oracle.rs

```diff
@@ -419,88 +417,102 @@ impl Poller {
     async fn fetch_product_and_price_accounts<'a, A>(
         &self,
         mapping_accounts: A,
-    ) -> Result<(
-        HashMap<Pubkey, ProductAccount>,
-        HashMap<Pubkey, PriceAccount>,
-    )>
+    ) -> Result<(HashMap<Pubkey, ProductEntry>, HashMap<Pubkey, PriceEntry>)>
     where
         A: IntoIterator<Item = &'a MappingAccount>,
     {
-        let mut pubkeys = vec![];
-        let mut futures = vec![];
+        let mut product_keys = vec![];
 
-        // Fetch all product accounts in parallel
+        // Get all product keys
         for mapping_account in mapping_accounts {
             for account_key in mapping_account
                 .products
                 .iter()
                 .filter(|pubkey| **pubkey != Pubkey::default())
             {
-                pubkeys.push(account_key.clone());
-                futures.push(self.fetch_product_account(account_key));
+                product_keys.push(account_key.clone());
             }
         }
 
-        let future_results = join_all(futures)
-            .await
-            .into_iter()
-            .collect::<Result<Vec<_>>>()?;
-
-        let product_accounts = zip(
-            pubkeys.into_iter(),
-            future_results
-                .clone()
-                .into_iter()
-                .map(|(product_account, _)| product_account),
-        )
-        .collect();
-
-        let price_accounts = future_results
-            .into_iter()
-            .flat_map(|(_, price_accounts)| price_accounts.into_iter())
-            .collect();
-
-        Ok((product_accounts, price_accounts))
-    }
+        // Look up all products with a single request
+        let product_accounts = self
+            .rpc_client
+            .get_multiple_accounts(product_keys.as_slice())
+            .await?;
```

Comment:
I am a bit worried that RPCs cannot handle this properly. Wdyt about batching it?

https://github.com/pyth-network/pyth-agent/actions/workflows/image-push.yaml 
Here we can dispatch a workflow from branch and with this I can test the new version and make sure this works.

Reply:
I think this is a good idea, I will add batching these requests. Users can always set this to an absurdly high value if their RPC can handle it.

Reply:
Done 🫡